### PR TITLE
Working deploy with travis-ci.com for the RaspberryPi3 (arm64) support

### PR DESCRIPTION
**Pull request by brainstorm**
This is a followup on the failed deploy for PR https://github.com/openpnp/openpnp-capture/pull/32.

The underlying issue has been resolved by travisci support folks, as explained/tracked on twitter:

https://twitter.com/braincode/status/1253981272885477376

This pullrequest should now deploy, as demonstrated on this fork:

https://github.com/brainstorm/openpnp-capture/releases
and
https://travis-ci.com/github/brainstorm/openpnp-capture/builds/162954136

As long as this particular repository is migrated to travis-ci.com instead of travis-ci.org (change to be operated by @vonnieda):

https://docs.travis-ci.com/user/migrate/open-source-repository-migration

### CODE REVIEW

**Comment by brainstorm** [comment]
Works on XCode 11.4.1 locally, although might need some tweaks on `platformcontext.mm` to avoid future deprecations:

```
$ make -j4
[ 36%] Building CXX object CMakeFiles/openpnp-capture.dir/common/libmain.cpp.o
[ 36%] Building CXX object CMakeFiles/openpnp-capture.dir/common/logging.cpp.o
[ 36%] Building CXX object CMakeFiles/openpnp-capture.dir/common/context.cpp.o
[ 36%] Building CXX object CMakeFiles/openpnp-capture.dir/common/stream.cpp.o
[ 45%] Building CXX object CMakeFiles/openpnp-capture.dir/mac/platformcontext.mm.o
[ 54%] Building CXX object CMakeFiles/openpnp-capture.dir/mac/platformstream.mm.o
[ 63%] Building CXX object CMakeFiles/openpnp-capture.dir/mac/uvcctrl.mm.o
/Users/romanvg/dev/cchs/openpnp-capture/mac/platformcontext.mm:74:54: warning: 'devicesWithMediaType:' is deprecated: first deprecated in macOS 10.15 - Use AVCaptureDeviceDiscoverySession instead. [-Wdeprecated-declarations]
    for (AVCaptureDevice* device in [AVCaptureDevice devicesWithMediaType:AVMediaTypeVideo])
                                                     ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/AVFoundation.framework/Headers/AVCaptureDevice.h:102:1: note: 'devicesWithMediaType:' has been explicitly marked deprecated here
+ (NSArray<AVCaptureDevice *> *)devicesWithMediaType:(AVMediaType)mediaType API_DEPRECATED("Use AVCaptureDeviceDiscoverySession instead.", ios(4.0, 10.0), macos(10.7, 10.15));
^
1 warning generated.
[ 72%] Linking CXX shared library libopenpnp-capture.dylib
[ 72%] Built target openpnp-capture
[ 81%] Building CXX object mac/tests/CMakeFiles/openpnp-capture-test.dir/__/__/common/logging.cpp.o
[ 90%] Building CXX object mac/tests/CMakeFiles/openpnp-capture-test.dir/main.cpp.o
[100%] Linking CXX executable openpnp-capture-test
[100%] Built target openpnp-capture-test
```

**Comment by brainstorm** [comment]
Oh dear, it turns out XCode 11.4 is fine, but TravisCI still does not support Catalina?:

https://travis-ci.community/t/macos-catalina-build-environment/5608/16

File: .travis.yml [hunk 1]
```diff
@@ -26,14 +36,16 @@ before_deploy:
   - mkdir deploy
   - if [ $TRAVIS_OS_NAME = osx ]; then cp libopenpnp-capture.dylib deploy; cp mac/tests/openpnp-capture-test deploy/openpnp-capture-test-mac; fi
   - if [ $TRAVIS_OS_NAME = linux ]; then cp libopenpnp-capture.so deploy; cp linux/tests/openpnp-capture-test deploy/openpnp-capture-test-linux; fi
-
+  - if [ $TRAVIS_CPU_ARCH = arm64 ]; then cp libopenpnp-capture.so deploy/libopenpnp-capture-arm64.so; cp linux/tests/openpnp-capture-test deploy/openpnp-capture-test-linux-arm64; fi
 deploy:
   provider: releases
-  api_key:
+  token:
```
Comment (vonnieda) [inline .travis.yml:42]: What is the purpose of this change? I don't see "token" documented on https://docs.travis-ci.com/user/deployment/releases/

File: .travis.yml [hunk 2]
```diff
@@ -26,14 +36,16 @@ before_deploy:
...
     secure: Vep5q/3rmKlU1bwi4R7/vrOOvOzZ/S+ukNBgjioMWMQP+f/QFZPc1IO0VP3kqxuxpvG4rWJR0djSu+FJG6ptK9OZbjVZvEAiBJVQZnHkDHyc8Gye2GCj/4XPUWiKXVQS5fTFHI7NenGPMZlRs/N9xI156gwiOLkpPK2Di1Zna51Mig74zU68WpDRsVWNlKu2nIx1WK2W7fWn9IM1FrtM86BPnztkqHUOBKhyYTpw7hKtAnqiB/09u5vBqd0DeoTohYaSAtFcKeNVBFtZO5wF+VKRnDDRMitM3SgEFElHT68KRR9zoqEvdmieK2GsoEhNzOKWRhjBWG1JvxJOlish8m1gftOiRPmNKq+PrLLReZpIRh7XoqXhmSW2M5BzrWnv1l957tRp0V/2dy6gZudCD4Kwh3CV/XXfUK6J6yPaCpCqSGV2ylHoX9hhVMheh4kdjpKUi4dsfQ5suLqgSIwge6trBjwyXUqrsnH9KAZJEpLCfIR4E870phOV5nuNkeFqS5mFdqJIlJPjORFzNgbJ5fx9yH9RqIHEuTH5Rkd8svPWKo7KU9L1wJcjm2k1VHqy5MCxenMGuRafUNAOQ9PoiAd7ijr/zvrUgvEjl353rWheTYllkRovt4C6DQ08fcC5w2BlofuRWWk/+Z69Pd0QkB7+dqae92csshaxe1TvAHA=
   file_glob: true
-  file: deploy/*    
-  skip_cleanup: true
+  file: deploy/*
+  cleanup: false
```
Comment (vonnieda) [inline .travis.yml:46]: Same as above: Why is this changed? "cleanup" does not seem to be documented.

File: .travis.yml [hunk 3]
```diff
@@ -26,14 +36,16 @@ before_deploy:
...
+  edge: true
```
Comment (vonnieda) [inline .travis.yml:47]: Okay, it looks like this selects the beta version of the provider, which explains the above questions. Is this required for your deployment to work? If it's not, I don't see a good reason to switch to a beta provider at this time.

File: .travis.yml [hunk 4]
```diff
@@ -26,14 +36,16 @@ before_deploy:
...
   on:
+    all_branches: true
```
Comment (vonnieda) [inline .travis.yml:49]: I don't think we want to create a new release for every branch. Is there some reason this is added?

Comment (brainstorm) [reply .travis.yml:49]: Yes, this was added since I was getting "Not deploying because `rpi3_arm64` is not an authorized branch". In my experience, I've found that when combined with `tags: true`, it conserves the original behaviour (only release on tags) and at the same time, it allows for easier deployment testing on forks (my case), since I didn't want to pollute my forked `master` to test whether it deployed well or not. Happy to remove it anyway now that the work is done.

File: .travis.yml (hunk 3 above)
Comment (brainstorm) [reply .travis.yml:47]: Yes, all the changes above are informed by TravisCI support, `travis lint`'s CLI and web linter too:

https://config.travis-ci.com/explore

See how your current `.travis.yml` on your `master` looks like against the official linter:

Skärmavbild_2020-05-07_kl__09_17_17

I really just wanted to future-proof this a little bit, but happy to revert back to the old syntax.

**Comment by vonnieda** [comment]
Thanks @brainstorm - all sounds good. I appreciate the detailed response. Let's give it a try!
